keymanager/src/client/remote.rs

```diff
@@ -584,4 +596,102 @@ impl KeyManagerClient for RemoteClient {
             .await
             .map_err(|err| KeyManagerError::Other(err.into()))
     }
+
+    async fn state_key(
+        &self,
+        churp_id: u8,
+        key_id: KeyPairId,
+    ) -> Result<StateKey, KeyManagerError> {
+        let id = (key_id, churp_id);
+
+        // First try to fetch from cache.
+        {
+            let mut cache = self.state_keys.write().unwrap();
+            if let Some(key) = cache.get(&id) {
+                return Ok(key.clone());
+            }
+        }
+
+        // No entry in cache, fetch from key manager.
+        let consensus_state = self.consensus_verifier.latest_state().await?;
+        let registry_state = consensus_state.registry_state();
+        let key_manager_state = consensus_state.key_manager_state();
+        let churp_state = key_manager_state.churp();
+
+        let status = tokio::task::block_in_place(move || {
+            let runtime_id = registry_state.key_manager(&self.runtime_id)?;
+            churp_state.status(runtime_id, churp_id)
+        })?;
+
+        let mut committee = status.committee;
+        committee.shuffle(&mut OsRng);
+
+        let kind = HandoffKind::CommitteeUnchanged;
+        let player = Player::new(status.threshold, kind);
+        let min_shares = player.min_shares();
+        let mut encoded_shares = Vec::with_capacity(min_shares);
+
+        // TODO: Optimize by fetching key shares concurrently.
```

Comment:
I think that two things:
- Peer feedback as you already mentioned.
- Support for multiple sessions with different peers+enclaves (currently only the server side handles multiple sessions, but not the client).

An ugly way to solve the last problem would be to create multiple clients. Which is not the best, as it would be nice to have a global client so that sessions can be shared if key manager methods are called from different places. And because of that, the session builder would need refactoring, as connections to the same node could use different policies and enclaves for session establishment.